### PR TITLE
Migrate to GitHub actions

### PR DESCRIPTION
**Pull request by joakimen**
This PR is an adaption of the CloudAssembly stuff previously done in Jenkins (see [Jenkinsfile](https://github.com/capralifecycle/liflig-repo-metrics/blob/master/Jenkinsfile)).

### Files changed
- Add `.ldp.json`
- Add `.github/workflows/ci.yml`
- Del `Jenkinsfile`

### Description

I have not tested the upload/trigger deploy stages, as I wanted to get more feedback at this point. The steps for creating the archive/metadata seem to work.

What the current steps do:
- `create cloud assembly archive`
  1. Create a cloud assembly zip-archive from the synthed snapshots
  2. Export archive path + s3 key (for upload stage)
- `create cloud assembly metadata file`
  1. Create a `cloud-assembly.json`-file containing casm archive S3 key + bucket name
  2. Export metadata file path + s3 key (for upload stage)
- `upload cloud assembly archive`: upload archive file using `upload-s3-file`
- `upload cloud assembly metadata file`: upload metadata file using `upload-s3-file`

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+        run: npm run test
+
+      - name: verify cdk snapshots
+        working-directory: ./packages/infrastructure
+        run: npm run snapshots && git status && git add -N && git diff --exit-code
+
+      - name: create cloud assembly archive
+        working-directory: ./packages/infrastructure
+        if: ${{ github.ref_name == github.event.repository.default_branch }}
```
Comment (stekern) [inline .github/workflows/ci.yml:66]: Depending on the type of trigger, `github.ref_name` isn't always set to the name of the branch. I believe `github.ref == format('refs/heads/{0}', github.event.repository.default_branch)` should be more robust

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      - uses: capralifecycle/actions-lib/check-runtime-dependencies@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+
+      - uses: capralifecycle/actions-lib/parse-config@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+        id: config
+        with:
+          config-file: ".ldp.json"
+
+      - uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8 # v4.0.1
+        with:
+          node-version: 16.20.2
```
Comment (stekern) [inline .github/workflows/ci.yml:36]: We should probably bump this

**Review by stekern — COMMENTED** [review]
Functionally this looks good!

There's an action `actions-lib/upload-cloud-assembly` that I suggest we use instead of doing creation and upload of the Cloud Assembly inline in this workflow. That action does not seem to be entirely complete, though - I think at least a `rm -rf cdk.out && npx cdk synth && cd cdk.out && zip -r "$INPUT_CLOUD_ASSEMBLY_ARCHIVE" .` is missing

File: .github/workflows/ci.yml (hunk 1 above)
Comment (joakimen) [reply .github/workflows/ci.yml:66]: Alright, will update, wasn't easy to find an authoritative answer on this one

File: .github/workflows/ci.yml (hunk 2 above)
Comment (joakimen) [reply .github/workflows/ci.yml:36]: Agreed, but the project doesn't build on Node 18 atm, and I didn't want to muddy up the GHA migration with lots of version bumping

**Comment by joakimen** [comment]
> Functionally this looks good!
> 
> There's an action `actions-lib/upload-cloud-assembly` that I suggest we use instead of doing creation and upload of the Cloud Assembly inline in this workflow. That action does not seem to be entirely complete, though - I think at least a `rm -rf cdk.out && npx cdk synth && cd cdk.out && zip -r "$INPUT_CLOUD_ASSEMBLY_ARCHIVE" .` is missing

Nice, I'll take a look at the missing stuff from the library before continuing this one

File: .github/workflows/ci.yml (hunk 2 above)
Comment (stekern) [reply .github/workflows/ci.yml:36]: It doesn't? It works locally with Node v20 at least

Comment (joakimen) [reply .github/workflows/ci.yml:36]: Yeah, works locally here too (both 18 and 20); I just reverted to the version it used previously after the errors that occurred during the build step [here](https://github.com/capralifecycle/liflig-repo-metrics/actions/runs/7756658205/job/21154415384). Might be unrelated, though

Comment (joakimen) [reply .github/workflows/ci.yml]: As previously noted, after having tried again with Node 20 which also seem to cause problems, we're handling the version bump separately from the migration.

**Review by stekern — APPROVED** [review]
LGTM! Nice work 🙂

**Pull request merged**